spec/services/contacts/filter_service_spec.rb

```diff
@@ -40,22 +43,37 @@
       it 'filter contacts by additional_attributes' do
         params[:payload] = payload
         result = filter_service.new(params, user_1).perform
-        expect(result.length).to be 2
+        expect(result[:contacts].length).to be 1
       end
 
       it 'filter contact by tags' do
         Contact.last.update_labels('support')
+        label_id = Contact.last.labels.last.id
         params[:payload] = [
           {
             attribute_key: 'labels',
             filter_operator: 'equal_to',
-            values: [1],
+            values: [label_id],
             query_operator: nil
           }.with_indifferent_access
         ]
 
         result = filter_service.new(params, user_1).perform
-        expect(result.length).to be 2
+        expect(result[:contacts].length).to be 1
```

Comment:
having 2-3 contacts with different labels.  
contact - 1  label.last
contact - 2 label.last, random label 
contact -  3 random label 

ensure the test returns contact 1 and 2 would help us to cover a lot more cases.